nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-jetty/src/main/java/org/apache/nifi/web/server/JettyServer.java

```diff
@@ -586,7 +586,14 @@ private WebAppContext loadWar(final File warFile, final String contextPath, fina
         // configure the max form size (3x the default)
         webappContext.setMaxFormContentSize(600000);
 
-        addHTTPHeaders(webappContext);
+        ArrayList<Class<? extends Filter>> filters = new ArrayList<>();
```

Comment:
I think you could do 

```final String ALL_PATHS = "/*";
List<Class<? extends Filter>> filters = new ArrayList<Class<? extends Filter>>(Arrays.asList(XFrameOptionsFilter.class, ContentSecurityPolicyFilter.class, XSSProtectionFilter.class));
if (props.isHTTPSConfigured()) {
    filters.add(StrictTransportSecurityFilter.class);
}
filters.forEach { (filter) -> addFilters(filter, ALL_PATHS, webappContext));```

That will create an `ArrayList` around the original `array` but allow for modification (_i.e. adding the STS filter if necessary_).

Reply:
Updated